chain/bitcoin/address.go

```diff
@@ -37,11 +39,12 @@ func NewAddressEncoder(params *chaincfg.Params) AddressEncoder {
 
 // EncodeAddress implements the address.Encoder interface
 func (encoder AddressEncoder) EncodeAddress(rawAddr address.RawAddress) (address.Address, error) {
+	// Validate that the base58 address is in fact in correct format.
 	encodedAddr := base58.Encode([]byte(rawAddr))
-	if _, err := btcutil.DecodeAddress(encodedAddr, encoder.params); err != nil {
-		// Check that the address is valid.
-		return address.Address(""), err
+	if _, err := btcutil.DecodeAddress(encodedAddr, &chaincfg.RegressionNetParams); err != nil {
+		return address.Address(""), fmt.Errorf("address validation error: %v", err)
```

Comment:
```suggestion
		return address.Address(""), err
```
The word "error" is redundant. Error message should be as concise as possible and provide the only context needed. In this case, the function only returns an error in one place, so there will be no ambiguity.